### DISCUSSION TITLE
Multiple Service Role Key Segmentation

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [supabase/supabase#15608] [Questions]
I have come across an issue related to service role keys. Is it possible to create new service role keys to segment roles at the service side? For example, I've been trying to create keys like `SERVICE_1_ROLE_KEY`, `SERVICE_2_ROLE_KEY`, etc., but I couldn't find any explicit instructions in the documentation. 

Related documentation: [Understanding API Keys
](https://supabase.com/docs/guides/api/api-keys)

**Comment by GaryAustin1** [comment]
You would need to into using other Postgres roles than bypass RLS if that is your goal.  Then use grants on tables/functions if you want each to do different things.  Since the role bypasses RLS you can't use RLS with a custom claim to differentiate.  Also as no matter what else you add, the jwt the role is still service_role, and they would all run as the service_role user in Postgres.

This discussion, especially towards the end has relevant discussion.  https://github.com/orgs/supabase/discussions/11948
